coordinator/src/metrics.rs

```diff
@@ -243,3 +257,48 @@ fn node_metrics(cx: &Context, inner_node: Arc<ln_dlc_node::node::Node<NodeStorag
         }
     }
 }
+
+pub fn collect_metrics_based_on_node_events(
```

Comment:
❓We had a conversation about this yesterday, but I would still like to hear from others on why we think collecting metrics like this is necessary if they are already stored in the database. Cc @bonomat.

I don't see much benefit and this code needs to be maintained.

Reply:
Context: @luckysori and me discussed if we *should* actually report this to prometheus at all. If we plan to use it for reporting we will created a report based on the database - what's the motivation to report it to prometheus?

Reply:
I see these two different types of _reports_: 
- business metrics to track KPIs: these are in the DB and are collected manually once every X (e.g. 1/week)
- liveness metrics: these are reported via prometheus to grafana so that we get a glimpse of what is happening.  

We can combine these: for that I would not send events around but rather have a recurring thread that polls from the db and collects the data and prepares it for prometheus/grafana. 
What do you think?

Reply:
I completely agree with the DB based reports. That makes sense.

I am not totally convinced about reporting this particular metric to prometheus. Could it be better to query the database from the Grafana board instead of reporting this via prometheus?

If we want to keep reporting this metric to prometheus I think it's better reporting it based on the event rather than a periodic task. If we have a periodic task that loads and reports fees it can happen that we miss reporting one (when we e.g. shut down the coordinator) unless we track what was already reported. This might create scenarios where we get confused because e.g. what is reported on prometheus does not match the report we got from the database. I am not convinced it's worth creating logic to track what we already reported; that's why I opted for reporting to prometheus based on the event.

Reply:
As discussed offline just now: We won't report `fees` as metrics to Grafana (through Prometheus) because there is nothing to act on here.